### PR TITLE
Added comments in conversation and dispatcher files

### PR DESCRIPTION
**Pull request by Priprocks**
Proposed Changes:
1) Added comments in conversation.py
2) Added comments and a description in dispatcher.py

**Status (please check what you already did)**:
- [ ] made PR ready for code review
- [ ] added some tests for the functionality
- [x] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

**Comment by Priprocks** [comment]
I do not understand.
There's double asterix missing in the _template_variables method, but when I made the change and pushed the commit, the travis build failed.

**Comment by tmbo** [comment]
Can you merge the latest master into your branch, we remove the topics, so there is some conflict here.

**Comment by Priprocks** [comment]
Done

**Comment by Priprocks** [comment]
Hey @tmbo ,
Actually I made a primitive working emulator for dialogflow.
I raised the issue about the same [here](https://github.com/RasaHQ/rasa_core/issues/589)
It would be helpful, if you check it once.
And should I commit to this PR or should wait for this to get merged and then create a PR for that?

**Comment by tmbo** [comment]
Sure thing, where is that?

File: rasa_core/conversation.py [hunk 1]
```diff
@@ -17,9 +17,34 @@ class Dialogue(object):
...
 
     def __str__(self):
+        # type: None -> Text
+
+        # This function returns the dialogue and turns.
         return "Dialogue with name '{}' and turns:\n{}".format(
                 self.name, "\n\n".join(["\t{}".format(t) for t in self.events]))
+
+
+class Topic(object):
```
Comment (tmbo) [inline rasa_core/conversation.py]: This class got removed on master and should not be readded. (this includes the code below till the end of the file)

**Comment by Priprocks** [comment]
Should I commit the emulator to this thread only or should I wait for this to get merged?

**Comment by tmbo** [comment]
Please don't use the same branch. Either use a different branch, or wait for this to get merged.

**Comment by Priprocks** [comment]
@tmbo did the changed. I'll create the PR for the emulator once this gets merged.

**Comment by tmbo** [comment]
Nice :rocket:

**Pull request merged**